### ISSUE TITLE
from_logits definition seems different from what is expected?

### ISSUE THREAD

**Issue by muralibalki** [apache/incubator-mxnet#12185]
In loss functions like SoftmaxCrossEntropy, the definition of from_logits is:
from_logits (bool, default False) – Whether input is a log probability (usually from log_softmax) instead of unnormalized numbers.

It seems mxnet calls the output of a log_softmax a logit, whereas others call (any) unscaled log probabilities logits.

e.g. In tensorflow/keras api:
e.g. https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits_v2
from_logits: Boolean, whether output is the result of a softmax, or is a tensor of logits.

I think mxnet is consistent with the definition of logits:
https://en.wikipedia.org/wiki/Logit

But highlighting the difference may be useful in general. I had a bit of trouble converting a model from Keras to Gluon because of this.

**Comment by vdantu** [comment]
@muralibalki : Thanks for pointing this out. 

@mxnet-label-bot: [Doc]

**Comment by zachgk** [comment]
@mxnet-label-bot add [Good First Issue]

@muralibalki As an apache open source project, we always welcome contributions from the community and our users! If you want to contribute, this would be a good issue to get started with. You can find help contributing to mxnet at https://mxnet.apache.org/community/contribute.html.

**Comment by braindotai** [comment]
[Check this out](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits)
Here it says "logits: Per-label activations, typically a linear output", which means `nd.dot(x, w) + b` in terms of MXNet. 
There are actually two versions of logits, first is simply the linear layer(as mentions in the above link), and second is the unscaled log probabilities. That is the reason Tensorflow provided 2 versions of "softmax_cross_entropy_with_logits".

In MXNet mx.gluon.loss.SoftmaxCrossEntropyLoss accepts the linear output(`nd.dot(x, w) + b`) as output in the argument. 
You can check [here](https://gluon.mxnet.io/chapter02_supervised-learning/softmax-regression-gluon.html), the layer definition is `net = gluon.nn.Dense(num_outputs)`,
defining the loss as `gluon.loss.SoftmaxCrossEntropyLoss()` and then calculating loss as
```python
output = net(data)
loss = softmax_cross_entropy(output, label)
```
